### PR TITLE
Draft 1 - Blogging!

### PR DESCRIPTION
**Pull request by kelvinma**

### CODE REVIEW

File: _posts/2016-12-20-sql-functions-vs-active-record.markdown [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        actual_holdback_amount,
+        advance_id
+      FROM payments p
+      WHERE p.payment_date >= payment_start_date AND p.payment_date <= payment_end_date
+    ) as p
+    ON p.advance_id = a.id
+    GROUP BY a.id
+```
+
+By structuring all of our subsequent calculations around this base query, we were able to bring down our total report time to around 27 seconds on the first iteration of the SQL report function. Further refactors to remove my myriad structural problems, mostly duplicated or redundant queries, I was finally able to get our current build down to... *drumroll please*
```
Comment (phereford) [inline _posts/2016-12-20-sql-functions-vs-active-record.markdown]: `my myriad structural problrems` -> `a myriad of structural problems` ?

File: _posts/2016-12-20-sql-functions-vs-active-record.markdown [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+EXPLAIN ANALYZE SELECT * FROM generate_report(~~PARAMS~~);
+                  QUERY PLAN                                                          
+--------------------------------------------
+ Function Scan on generate_report  (cost=0.25..10.25 rows=1000 width=1552) (actual time=3610.526..3610.817 rows=2995 loops=1)
+ Planning time: 0.087 ms
+ Execution time: 3611.286 ms
+(3 rows)
+```
+
+Implementing the SQL query in to our existing Rails stack involved a trivial amount of overhead, setting up a sanitized query via Rails `sanitize_sql_for_conditions` and building out a custom report record `Struct` to handle the query table.
```
Comment (phereford) [inline _posts/2016-12-20-sql-functions-vs-active-record.markdown]: lets link out to rails docs for `sanitize_sql_for_conditions`.

Comment (kelvinma) [inline _posts/2016-12-20-sql-functions-vs-active-record.markdown]: 👍

File: _posts/2016-12-20-sql-functions-vs-active-record.markdown (hunk 1 above)
Comment (kelvinma) [inline _posts/2016-12-20-sql-functions-vs-active-record.markdown]: It's myriad. Time for some fun grammar trivia...

Myriad in Greek is 10,000. 

Thus:
>10,000 men == Myriad men
>10,000 of men == Myriad of men

Technically, both usages of `myriad men` and `myriad of men` are correct, but the more literally elegant usage is always as an adjective. Most copy editors will change it to the adjective usage.

**Pull request merged**